Lib/test/test_repl.py

```diff
@@ -107,6 +107,26 @@ def test_close_stdin(self):
         self.assertEqual(process.returncode, 0)
         self.assertIn('before close', output)
 
+    def test_interactive_traceback_reporting(self):
```

Comment:
Nice!

I think we should move this to a separate class, so we can add more syntax errors to this in the future, maybe even making the method something where you can give the expected string and the code directly.

What do you think?

Reply:
Agreed. I've just copied off the same solution from #27117 but a proper interface sure simplifies a lot (and would be helpful for future bug reports)!

Reply:
Yeah, I would prefer to avoid having many different methods for what is at the end just testing different input and output.